src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -504,6 +505,17 @@ private sealed class FileSystemEnumerable<T> : IEnumerable<T>
                         }
                         searchPattern = searchPattern.Substring(lastSlash + 1);
                     }
+
+                    // Typically we shouldn't see either of these cases
+                    if (searchPattern.IndexOfAny(s_EscapeCharacters) != -1)
```

Comment:
Yes, dealing with the string replace does a bit more work. Running through a quick test on 64 bit CoreFX it was over twice as fast (note that the old 32 bit jitter was only 10% faster). Doing a manual loop, however, was nearly 5 times as fast. Switching to that.